### PR TITLE
Fix ReferenceTrimmer on Linux

### PR DESCRIPTION
**Pull request by stan-sz**
Fixes the `Sequence contains no matching element` observed on Linux machines.

### CODE REVIEW

File: src/ReferenceTrimmerTask.cs [hunk 1]
```diff
@@ -185,7 +185,7 @@ public override bool Execute()
                     .Where(path => !path.EndsWith("_._", StringComparison.Ordinal)) // Ignore special packages
                     .Select(path =>
                     {
-                        var packageFolderRelativePath = Path.Combine(nugetLibrary.Name, nugetLibrary.Version.ToNormalizedString(), path);
+                        var packageFolderRelativePath = Path.Combine(nugetLibrary.Name.ToLowerInvariant(), nugetLibrary.Version.ToNormalizedString(), path);
```
Comment (dfederm) [inline src/ReferenceTrimmerTask.cs:188]: Can you add a code comment explaining that this is what NuGet does? Ideally if you can find the place it does it that would be nice, but not necessary.

Comment (stan-sz) [reply src/ReferenceTrimmerTask.cs:188]: Seems like the ToLowerInvariant also needs to be applied to version

Comment (stan-sz) [reply src/ReferenceTrimmerTask.cs:188]: @jeffkl - what's the best way to mimic NuGet logic here to find the location of package artifacts in an OS-agnostic manner?

Comment (jeffkl) [reply src/ReferenceTrimmerTask.cs:188]: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/VersionFolderPathResolver.cs#L201

**Review by jeffkl — APPROVED** [review]

File: src/ReferenceTrimmerTask.cs (hunk 1 above)
Comment (stan-sz) [reply src/ReferenceTrimmerTask.cs:188]: The package path information is available in the lockfile, which is the crux of the latest iteration.

**Pull request merged**